### ISSUE TITLE
Assigning to *args causes a compile error

### ISSUE THREAD

**Issue by johnw-bluemark** [agronholm/typeguard#330]
### Things to check first

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Typeguard version

3.0.2

### Python version

3.8.10, 3.10.11

### What happened?

It errors during compile. It should not error during compile

### How can we reproduce the bug?

test.py
```python
def test(*args):
    args = []
```
typecheck.py
```python
from typeguard import install_import_hook
install_import_hook('test')
import test
```

run typecheck.py

Python 3.8:
```
Traceback (most recent call last):
  File "typecheck.py", line 3, in <module>
    import test
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "C:\Python38\lib\site-packages\typeguard\_importhook.py", line 96, in exec_module
    super().exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 844, in exec_module
  File "<frozen importlib._bootstrap_external>", line 981, in get_code
  File "C:\Python38\lib\site-packages\typeguard\_importhook.py", line 85, in source_to_code
    return _call_with_frames_removed(
  File "C:\Python38\lib\site-packages\typeguard\_importhook.py", line 42, in _call_with_frames_removed
    return f(*args, **kwargs)
TypeError: expected some sort of slice, but got <_ast.Tuple object at 0x000001FC89BD5370>
```

Python 3.10:
```
Traceback (most recent call last):
  File "C:\Users\Johnw\Desktop\temp\typecheck.py", line 3, in <module>
    import test
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "C:\Python310\lib\site-packages\typeguard\_importhook.py", line 96, in exec_module
    super().exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "C:\Python310\lib\site-packages\typeguard\_importhook.py", line 85, in source_to_code
    return _call_with_frames_removed(
  File "C:\Python310\lib\site-packages\typeguard\_importhook.py", line 42, in _call_with_frames_removed
    return f(*args, **kwargs)
TypeError: required field "ctx" missing from Subscript
```

**Comment by agronholm** [comment]
This works with 4.0.0rc1. I plan to release the final version soon, as long as there are no showstopper issues.

**Comment by johnw-bluemark** [comment]
So it does. I'm now testing 4.0.0.rc1, and I've encountered a different error in a large project. I'll make a new bug report when I can narrow down the cause.

**Comment by johnw-bluemark** [comment]
Actually, this issue still happens on Python 3.10.11 with 4.0.0rc1.post2

**Comment by agronholm** [comment]
Make sure you delete any `__pycache__` directories so that the compilation is done again and not loaded from cache.

**Comment by johnw-bluemark** [comment]
The 4.0.0.rc1 issue was indeed related to `__pycache__`.

**Comment by agronholm** [comment]
Yeah. I'm adding a mechanism to typeguard that varies the file name pattern of the generated typeguard-modified .pyc files based on the version information obtained from `importlib.metadata` (if available).

**Comment by johnw-bluemark** [comment]
I've found a very similar issue in 4.0.0.rc1

test.py
```python
def test(*args: str):
    pass
```

Giving any type annotation to *args causes the same error I reported above, but only for Python 3.8. Python 3.10 runs it just fine.

**Comment by agronholm** [comment]
I've added a new test for both positional and keyword variable arguments, and indeed it fails on Python 3.7 and 3.8. Good thing this was caught before the final release.

**Comment by basnijholt** [comment]
I also keep running into that issue in https://github.com/python-adaptive/adaptive/. 

> This works with 4.0.0rc1. I plan to release the final version soon, as long as there are no showstopper issues.

What is your idea of "soon"? 😄

**Comment by agronholm** [comment]
My plan was to release it next week, provided that no other showstopper issues crop up.

**Comment by agronholm** [comment]
So, turns out that on Python 3.8 and earlier, you can't assign `ast.Tuple` to the `slice` field of `ast.Subscript`, you have to wrap it in `ast.Index`.

**Comment by agronholm** [comment]
The fix is available in v4.0.0rc2.
